evalml/automl/automl_search.py

```diff
@@ -665,7 +677,15 @@ def _compute_cv_scores(self, pipeline, X, y):
 
     def _add_result(self, trained_pipeline, parameters, training_time, cv_data, cv_scores):
         cv_score = cv_scores.mean()
-        percent_better = self.objective.calculate_percent_difference(cv_score, self._baseline_cv_score)
+
+        percent_better_than_baseline = {}
+        mean_cv_all_objectives = self._get_mean_cv_scores_for_all_objectives(cv_data)
+        for obj_name in mean_cv_all_objectives:
+            objective_class = get_objective(obj_name)
+            percent_better = objective_class.calculate_percent_difference(mean_cv_all_objectives[obj_name],
+                                                                          self._baseline_cv_scores[obj_name])
+            percent_better_than_baseline[obj_name] = percent_better
```

Comment:
@gsheni This is what it would look like in `automl.results` :

```python
'percent_better_than_baseline_all_objectives': {'Log Loss Binary': 98.9276200826952,
    'MCC Binary': nan,
    'AUC': 98.17222663713376,
    'Precision': nan,
    'F1': nan,
    'Balanced Accuracy Binary': 91.62898962401862,
    'Accuracy Binary': 53.49893478518168}
```

This is slightly different from what you posted in the original issue but I think it's more human-readable. Happy to change though.

Reply:
- The reasoning for a list of dictionaries is that it would allow for adding more information in the future.
- For example, EvalML could calculate a boolean that looks at the baseline and pipeline score to determine if its actually better (even if `baselinePercentChange` is NaN).
- However, this is a major sticking point, and the end-user could get the scores, make the comparison, and calculate the boolean. 
 
```json
'percent_better_than_baseline_all_objectives': [
   {'objective' : 'Log Loss Binary',
    'baselinePercentChange': 98.9276200826952,
    'isBetterThanBaseline': True},
   {'objective' : 'MCC Binary',
    'baselinePercentChange': nan,
    'isBetterThanBaseline': True}...}
```

Reply:
Looks good to me!

Yep. Users can certainly easily compute that boolean if they want to.